### PR TITLE
Adds selectPlatform method for amp-subscriptions

### PR DESCRIPTION
**Pull request by prateekbh**
- Adds `selectPlatform` method in entitlements store
- Implements `allEntitlementsResolved` promise

### CODE REVIEW

File: extensions/amp-subscriptions/0.1/entitlement-store.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ export class EntitlementStore {
         }
       }
 
-      if (entitlementsResolved === this.serviceIds_.length) {
+      if (this.areAllPlatformsResolved_()) {
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/entitlement-store.js]: That means this promise may never resolve if it wasn't resolved yet. You need to check `areAllplatformsResolved_` in the `onChange` below as well. That should be covered by a test as well.

File: extensions/amp-subscriptions/0.1/entitlement-store.js [hunk 2]
```diff
@@ -101,10 +101,65 @@ export class EntitlementStore {
...
+    return this.allResolvedPromise_;
+  }
+
+  /**
+   * Returns entitlements when all services are done fetching them.
+   * @returns {!Promise<!Entitlements>}
+   */
+  selectPlatform() {
+    return this.getAllPlatformsEntitlements_().then(entitlements => {
+      return this.selectApplicablePlatform_(entitlements);
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/entitlement-store.js]: Add a `TODO` to explain that in the future we may be able to resolve a selected platform faster than waiting for all platforms. E.g. a first "subscriber" platform can be selected right away.

File: extensions/amp-subscriptions/0.1/entitlement-store.js (hunk 1 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/entitlement-store.js]: Done!

File: extensions/amp-subscriptions/0.1/entitlement-store.js (hunk 2 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/entitlement-store.js]: Done

**Pull request merged**